app/handlers/handlers_connectors.py

```diff
@@ -74,11 +74,39 @@ def get_connector_pod(
                     "seccompProfile": {
                         "type": "RuntimeDefault"
                     },
-               },
+                    "readOnlyRootFilesystem": True,
+                },
+                "readinessProbe": {
+                    "exec": {
+                        "command": [
+                            "/connectorctl",
+                            "health",
+                        ]
+                    },
+                    "initialDelaySeconds": 5,
+                    "periodSeconds": 5,
+                },
+                "livenessProbe": {
+                    "exec": {
+                        "command": [
+                            "/connectorctl",
+                            "health",
+                        ]
+                    },
+                    "initialDelaySeconds": 5,
+                    "periodSeconds": 5,
+                },
+                "volumeMounts": [
+                    {
+                        "name": "twingate-socket",
+                        "mountPath": "/var/run/twingate",
+                    }
+                ],
                 **container_extra,
             },
             *spec.sidecar_containers,
         ],
+        "volumes": [{"name": "twingate-socket", "emptyDir": {}}],
         **spec.pod_extra,
```

Comment:
I'm not 100% sure about this but I think we should pop the `volumes` key from `spec.pod_extra` dict and `volumeMounts` from `spec.container_extra` because a user might specify those keys and unintentionally break the health checks when `readOnlyRootFilesystem` is true. @ekampf WDYT?

Reply:
We need to support `pod_extra` having other volumes